tuner/tuner/op_matchers.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
```

Comment:
Lots of boiler plate here, but this basically implements matching for a generic op with MMT indexing maps. The important functions of the matchers are `match` and `get_matched_ops`, which match a specific op, or get all the ops in a parent operation that match.